es_source.go

```diff
@@ -133,13 +152,44 @@ func boundsFilter(geometryField string, tile maptile.Tile) *Dict {
 	}
 }
 
+// Given the list of extra source arguments that were specified with request, transform
+// these into a map of property name to ES document source path, or return an error
+// if there is a malformed extra source argument.
+func makeFieldMap(incArgs []string) (map[string]string, error) {
+	var result = make(map[string]string);
+
+	for _, source := range incArgs {
+		splits := strings.SplitN(source, ":", 2);
+		if len(splits) < 2 {
+			return nil, InvalidRequestError{fmt.Sprintf("Invalid source field specification: '%s'", source)};
+		}
+		result[splits[0]] = splits[1];
+	}
+
+	return result, nil;
+}
+
 // doGetFeatures scrolls the configured Elasticsearch index for all documents that fall
 // within the tile boundaries
 func (e *ElasticsearchSource) doGetFeatures(ctx context.Context, req *TileRequest) (*geojson.FeatureCollection, error) {
+	// Check for optional ES query argument.
 	var query = elastic.NewBoolQuery().Filter(boundsFilter(e.GeometryField, req.MapTile()))
-	if qs, exists := req.Args["q"]; exists && qs != "" {
-		query = query.Filter(elastic.NewQueryStringQuery(qs))
+	if qs, exists := req.Args["q"]; exists && qs[0] != "" { // TODO: We ignore all but the first "q" arg.
+		query = query.Filter(elastic.NewQueryStringQuery(qs[0]))
```

Comment:
Minor Go-ism here: the original code checked for `qs != ""` because in Go, a `string`'s default/null-value is `""`. In this case, since `qs` is now a `[]string`, you should instead be checking for `len(qs) > 0`.

Reply:
OIC. -- Not sure if it is possible to somehow pas the empty string as `q` argument (vs. omitting it), but I guess that's fine b/c we expect the caller to pass in something that is a query that ES can make sense of.